src/hexen/sc_man.c

```diff
@@ -197,16 +197,16 @@ boolean SC_GetString(void)
     {
         while (*ScriptPtr <= 32)
         {
-            if (ScriptPtr >= ScriptEndPtr)
-            {
-                sc_End = true;
-                return false;
-            }
             if (*ScriptPtr++ == '\n')
             {
                 sc_Line++;
                 sc_Crossed = true;
             }
+            if (ScriptPtr >= ScriptEndPtr)
```

Comment:
Just change the `while` condition above to:
```
  while (ScriptPtr < ScriptEndPtr && *ScriptPtr <= 32)
```
and this block is no longer needed.